<a href="https://colab.research.google.com/github/Kaustubh0505/C_G15_dvaCapstone/blob/main/notebooks/01_extraction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [31]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

In [32]:
df = pd.read_csv("zomato.csv", encoding="latin-1")

In [33]:
df.shape

(51717, 17)

In [34]:
df.columns

Index(['url', 'address', 'name', 'online_order', 'book_table', 'rate', 'votes',
       'phone', 'location', 'rest_type', 'dish_liked', 'cuisines',
       'approx_cost(for two people)', 'reviews_list', 'menu_item',
       'listed_in(type)', 'listed_in(city)'],
      dtype='object')

In [35]:
df.head()

,url,address,name,online_order,book_table,rate,votes,phone,location,rest_type,dish_liked,cuisines,approx_cost(for two people),reviews_list,menu_item,listed_in(type),listed_in(city)
0,https://www.zomato.com/bangalore/jalsa-banasha...,"942, 21st Main Road, 2nd Stage, Banashankari, ...",Jalsa,Yes,Yes,4.1/5,775,080 42297555\r\n+91 9743772233,Banashankari,Casual Dining,"Pasta, Lunch Buffet, Masala Papad, Paneer Laja...","North Indian, Mughlai, Chinese",800,"[('Rated 4.0', 'RATED\n A beautiful place to ...",[],Buffet,Banashankari
1,https://www.zomato.com/bangalore/spice-elephan...,"2nd Floor, 80 Feet Road, Near Big Bazaar, 6th ...",Spice Elephant,Yes,No,4.1/5,787,080 41714161,Banashankari,Casual Dining,"Momos, Lunch Buffet, Chocolate Nirvana, Thai G...","Chinese, North Indian, Thai",800,"[('Rated 4.0', 'RATED\n Had been here for din...",[],Buffet,Banashankari
2,https://www.zomato.com/SanchurroBangalore?cont...,"1112, Next to KIMS Medical College, 17th Cross...",San Churro Cafe,Yes,No,3.8/5,918,+91 9663487993,Banashankari,"Cafe, Casual Dining","Churros, Cannelloni, Minestrone Soup, Hot Choc...","Cafe, Mexican, Italian",800,"[('Rated 3.0', ""RATED\n Ambience is not that ...",[],Buffet,Banashankari
3,https://www.zomato.com/bangalore/addhuri-udupi...,"1st Floor, Annakuteera, 3rd Stage, Banashankar...",Addhuri Udupi Bhojana,No,No,3.7/5,88,+91 9620009302,Banashankari,Quick Bites,Masala Dosa,"South Indian, North Indian",300,"[('Rated 4.0', ""RATED\n Great food and proper...",[],Buffet,Banashankari
4,https://www.zomato.com/bangalore/grand-village...,"10, 3rd Floor, Lakshmi Associates, Gandhi Baza...",Grand Village,No,No,3.8/5,166,+91 8026612447\r\n+91 9901210005,Basavanagudi,Casual Dining,"Panipuri, Gol Gappe","North Indian, Rajasthani",600,"[('Rated 4.0', 'RATED\n Very good restaurant ...",[],Buffet,Banashankari


In [36]:
df.tail()

,url,address,name,online_order,book_table,rate,votes,phone,location,rest_type,dish_liked,cuisines,approx_cost(for two people),reviews_list,menu_item,listed_in(type),listed_in(city)
51712,https://www.zomato.com/bangalore/best-brews-fo...,"Four Points by Sheraton Bengaluru, 43/3, White...",Best Brews - Four Points by Sheraton Bengaluru...,No,No,3.6 /5,27,080 40301477,Whitefield,Bar,NaN,Continental,"1,500","[('Rated 5.0', ""RATED\n Food and service are ...",[],Pubs and bars,Whitefield
51713,https://www.zomato.com/bangalore/vinod-bar-and...,"Number 10, Garudachar Palya, Mahadevapura, Whi...",Vinod Bar And Restaurant,No,No,NaN,0,+91 8197675843,Whitefield,Bar,NaN,Finger Food,600,[],[],Pubs and bars,Whitefield
51714,https://www.zomato.com/bangalore/plunge-sherat...,Sheraton Grand Bengaluru Whitefield Hotel & Co...,Plunge - Sheraton Grand Bengaluru Whitefield H...,No,No,NaN,0,NaN,Whitefield,Bar,NaN,Finger Food,"2,000",[],[],Pubs and bars,Whitefield
51715,https://www.zomato.com/bangalore/chime-sherato...,Sheraton Grand Bengaluru Whitefield Hotel & Co...,Chime - Sheraton Grand Bengaluru Whitefield Ho...,No,Yes,4.3 /5,236,080 49652769,"ITPL Main Road, Whitefield",Bar,"Cocktails, Pizza, Buttermilk",Finger Food,"2,500","[('Rated 4.0', 'RATED\n Nice and friendly pla...",[],Pubs and bars,Whitefield
51716,https://www.zomato.com/bangalore/the-nest-the-...,"ITPL Main Road, KIADB Export Promotion Industr...",The Nest - The Den Bengaluru,No,No,3.4 /5,13,+91 8071117272,"ITPL Main Road, Whitefield","Bar, Casual Dining",NaN,"Finger Food, North Indian, Continental","1,500","[('Rated 5.0', 'RATED\n Great ambience , look...",[],Pubs and bars,Whitefield


In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51717 entries, 0 to 51716
Data columns (total 17 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   url                          51717 non-null  object
 1   address                      51717 non-null  object
 2   name                         51717 non-null  object
 3   online_order                 51717 non-null  object
 4   book_table                   51717 non-null  object
 5   rate                         43942 non-null  object
 6   votes                        51717 non-null  int64 
 7   phone                        50509 non-null  object
 8   location                     51696 non-null  object
 9   rest_type                    51490 non-null  object
 10  dish_liked                   23639 non-null  object
 11  cuisines                     51672 non-null  object
 12  approx_cost(for two people)  51371 non-null  object
 13  reviews_list                 51

### Data Type Observations

- Most columns are of type 'object', indicating categorical/text data
- 'votes' is the only numeric column (int64)
- 'rate' should ideally be numeric but is stored as object (likely due to values like 'NEW', '-')
- 'approx_cost(for two people)' is also stored as object and will need conversion to numeric

In [38]:
df.describe()

,votes
count,51717.000000
mean,283.697527
std,803.838853
min,0.000000
25%,7.000000
50%,41.000000
75%,198.000000
max,16832.000000


In [39]:
df.isnull().sum()

,0
url,0
address,0
name,0
online_order,0
book_table,0
rate,7775
votes,0
phone,1208
location,21
rest_type,227


### Missing Values Observation

- 'rate' column has significant missing values (~7800)
- 'dish_liked' has very high missing values (~28000+) → may not be reliable
- 'phone' also has missing entries
- 'approx_cost(for two people)' has some missing values

In [40]:
df.duplicated().sum()

np.int64(0)

In [41]:
df['rate'].unique()

array(['4.1/5', '3.8/5', '3.7/5', '3.6/5', '4.6/5', '4.0/5', '4.2/5',
       '3.9/5', '3.1/5', '3.0/5', '3.2/5', '3.3/5', '2.8/5', '4.4/5',
       '4.3/5', 'NEW', '2.9/5', '3.5/5', nan, '2.6/5', '3.8 /5', '3.4/5',
       '4.5/5', '2.5/5', '2.7/5', '4.7/5', '2.4/5', '2.2/5', '2.3/5',
       '3.4 /5', '-', '3.6 /5', '4.8/5', '3.9 /5', '4.2 /5', '4.0 /5',
       '4.1 /5', '3.7 /5', '3.1 /5', '2.9 /5', '3.3 /5', '2.8 /5',
       '3.5 /5', '2.7 /5', '2.5 /5', '3.2 /5', '2.6 /5', '4.5 /5',
       '4.3 /5', '4.4 /5', '4.9/5', '2.1/5', '2.0/5', '1.8/5', '4.6 /5',
       '4.9 /5', '3.0 /5', '4.8 /5', '2.3 /5', '4.7 /5', '2.4 /5',
       '2.1 /5', '2.2 /5', '2.0 /5', '1.8 /5'], dtype=object)

In [42]:
df['online_order'].unique()

array(['Yes', 'No'], dtype=object)

In [43]:
df['book_table'].unique()

array(['Yes', 'No'], dtype=object)

In [44]:
df.sample(5)

,url,address,name,online_order,book_table,rate,votes,phone,location,rest_type,dish_liked,cuisines,approx_cost(for two people),reviews_list,menu_item,listed_in(type),listed_in(city)
28720,https://www.zomato.com/bangalore/the-burger-pl...,"2nd Floor, MMR Plaza, Above DCB Bank, Sarjapur...",The Burger Place,Yes,No,3.8 /5,28,+91 9108974600,Koramangala 1st Block,Quick Bites,NaN,"Burger, Continental, Fast Food",400,"[('Rated 4.0', ""RATED\n I was a little scepti...","['Signature Burger', 'Falafel Burger', 'Crunch...",Dine-out,Koramangala 4th Block
5644,https://www.zomato.com/bangalore/ambur-biryani...,"64, St Johns Church Road, Frazer Town, Bangalore",Ambur Biryani,Yes,No,NaN,0,+91 9830199065,Frazer Town,Quick Bites,NaN,"Biryani, Kebab",300,[],[],Delivery,Brigade Road
32949,https://www.zomato.com/bangalore/bawarchi-inn-...,"1, 1st Floor, BTM Complex, 9th Main Road, 1st ...",Bawarchi Inn,Yes,No,3.1 /5,140,+91 8197022223\r\n+91 9032216666,BTM,Casual Dining,"Dum Biryani, Chicken Biryani, Rice, Roti, Mutt...","North Indian, Biryani, Andhra, Chinese",600,"[('Rated 4.0', 'RATED\n delivery was late due...",[],Delivery,Koramangala 6th Block
8025,https://www.zomato.com/bangalore/rolls-on-whee...,"18, R J Garden, Outer Ring Road, Marathahalli,...",Rolls On Wheels,No,No,3.6/5,255,+91 9071734110\r\n+91 9071496800,Marathahalli,Quick Bites,"Rolls, Mix Veg","Fast Food, Rolls, Momos",200,"[('Rated 3.0', 'RATED\n I ordered cheesy jala...",[],Dine-out,Brookefield
27315,https://www.zomato.com/bangalore/hotel-vkm-btm...,"3/5, Opposite Forum Mall, Hosur Cross Road, BT...",HOTEL VKM,Yes,No,NaN,0,+91 9880622209\r\n+91 7411290003,BTM,Quick Bites,NaN,South Indian,300,"[('Rated 3.0', 'RATED\n Wanted to have a quic...","['Chicken Fried Rice', 'Chicken Kabab', 'Chill...",Delivery,Koramangala 4th Block


### Initial Observations

- Dataset contains 51,717 rows and 17 columns
- No duplicate records found
- Significant missing values present in 'dish_liked' and 'rate'
- 'rate' and 'approx_cost(for two people)' require data type conversion
- Majority of columns are categorical (object type)
- Some columns like 'url', 'phone', and 'address' may not be useful for analysis
- Presence of inconsistent values in 'rate' column (e.g., 'NEW', '-')